core/tests/test_stream.h

```diff
@@ -0,0 +1,239 @@
+// -------------------------------------------------------------
+//  Cubzh Core Unit Tests
+//  test_stream.h
+//  Created by Xavier Legland on October 21, 2022.
+// -------------------------------------------------------------
+
+#pragma once
+
+#include "stream.h"
+
+// functions that are NOT tested:
+// stream_new_buffer_write
+// stream_free
+// stream_new_buffer_write_prealloc
+// stream_buffer_unload
+// stream_new_file_write
+
+// check that we read the content correctly
+void test_stream_new_buffer_read(void) {
+	const size_t len = 6;
+	const char* content = "Hello";
+	Stream* s = stream_new_buffer_read(content, len);
+	char* buf = (char*)malloc(len * sizeof(char));
+	stream_read_string(s, len, buf);
+
+	TEST_CHECK(strcmp(buf, content) == 0);
+
+	stream_free(s);
+	free(buf);
+}
+
+// check that the file is read correctly
+void test_stream_new_file_read(void) {
+	const char* file_name = "hi.txt";
+	const size_t len = 6;
+	const char content[6] = "Hello";
+	char* buf = (char*)malloc(len * sizeof(char));
+	FILE* f = fopen(file_name, "w"); // open for writing
+	const int err = fputs(content, f);
+	TEST_ASSERT(err != EOF);
+	fclose(f);
+	f = fopen(file_name, "r");	     // open for reading
+	Stream* s = stream_new_file_read(f);
+	stream_read_string(s, 5, buf);
+	buf[len-1] = '\0';               // end the string
+
+	TEST_CHECK(strcmp(buf, content) == 0);
+
+	stream_free(s);
+	free(buf);
+	remove(file_name);
+}
+
+// check that the output matches the content
+void test_stream_read(void) {
+	const size_t len = 6;
+	char* content = (char*)malloc(len * sizeof(char));
+	content = "Hello";
```

Comment:
it doesn't work like that.
we can talk about it.

If I understand correctly, `strcpy` must be used here.

Also, aren't you getting a warning regarding assigning a `const char*` value to a `char*` variable?

Reply:
No warning on mvs 2022